Source/WebCore/platform/audio/cocoa/AudioDecoderCocoa.cpp

```diff
@@ -0,0 +1,349 @@
+
+        if (codec == kAudioFormatFLAC && config.description.empty())
+            return "Decoder config description for flac codec is mandatory"_s;
+
+        if (codec == 'vorb' && config.description.empty())
+            return "Decoder config description for vorbis codec is mandatory"_s;
+
+        mIsAAC = codec == kAudioFormatMPEG4AAC || codec == kAudioFormatMPEG4AAC_HE || codec == kAudioFormatMPEG4AAC_LD || codec == kAudioFormatMPEG4AAC_HE_V2 || codec == kAudioFormatMPEG4AAC_ELD;
+
+        AudioStreamBasicDescription absd { };
+        absd.mFormatID = codec;
+        // Attempt to create description with provided cookie.
+        bool succeeded = false;
+        if (config.description.size()) {
+            UInt32 size = sizeof(absd);
+            succeeded = PAL::AudioFormatGetProperty(kAudioFormatProperty_FormatInfo, config.description.size(), config.description.data(), &size, &absd) == noErr;
+        }
+        if (!succeeded) {
+            absd.mSampleRate = double(config.sampleRate);
+            absd.mChannelsPerFrame = uint32_t(config.numberOfChannels);
+        }
+        m_inputDescription = absd;
+    } else
+        m_inputDescription = CAAudioStreamDescription { double(config.sampleRate), uint32_t(config.numberOfChannels), *format, CAAudioStreamDescription::IsInterleaved::Yes };
+
+    m_inputFormatDescription = createAudioFormatDescription(*m_inputDescription, m_codecDescription.span());
+
+    m_outputDescription = CAAudioStreamDescription { double(config.sampleRate), uint32_t(config.numberOfChannels), AudioStreamDescription::Float32, CAAudioStreamDescription::IsInterleaved::No };
+
+    AudioSampleBufferConverter::Options options = {
+        .format = kAudioFormatLinearPCM,
+        .description = m_outputDescription->streamDescription(),
+        .generateTimestamp = false
+    };
+
+    m_converter = AudioSampleBufferConverter::create(decompressedAudioOutputBufferCallback, this, options);
+    if (!m_converter)
+        return "Couldn't create AudioSampleBufferConverter"_s;
+
+    return { };
+}
+
+Ref<SharedBuffer> InternalAudioDecoderCocoa::stripADTSHeader(SharedBuffer& buffer)
+{
+    static constexpr int kADTSHeaderSize = 7;
+    if (buffer.size() < kADTSHeaderSize)
+        return buffer;
+
+    auto matchesSync = [](std::span<const uint8_t> data) {
+        return data.size() >= 2 && data[0] == 0xff && (data[1] & 0xf6) == 0xf0;
+    };
+
+    if (!matchesSync(buffer.span()))
+        return buffer;
+
+    return buffer.getContiguousData(kADTSHeaderSize, buffer.size() - kADTSHeaderSize);
+}
+
+Ref<AudioDecoder::DecodePromise> InternalAudioDecoderCocoa::decode(Ref<SharedBuffer>&& frameData, [[maybe_unused]] bool isKeyFrame, int64_t timestamp, std::optional<uint64_t> duration)
+{
+    LOG(Media, "Decoding%s frame", isKeyFrame ? " key" : "");
+
+    if (m_isClosed)
+        return AudioDecoder::DecodePromise::createAndReject("Decoder is closed"_s);
```

Comment:
How about `using DecodePromise = AudioDecoder::DecodePromise;` to shorten these lines?